transcom/mymove
# INT I-12832 and B-19474 fix and enhancements

## [Issue ticket](https://www13.v1host.com/USTRANSCOM38/Issue.mvc/Summary?oidToken=Issue%3A944686)

[INT PR for B-19474](https://github.com/transcom/mymove/pull/12464)

## Summary

Couple of things going down in this PR. It was found when testing B-19474 that when the Prime updated the estimated weight of a shipment AFTER creation, then the status of the move was changing to `Approvals Requested` despite no approval or action needed from the TOO.

In order to fix this, I've updated the existing logic to check some values before changing the status that match the requirements of showing the action that the TOO needs to acknowledge the MTO is at risk of excess weight (the total of all shipments exceeding 90% of allowable weight).
- After the backend checks for excess weight, we check if `excess_weight_qualified_at` (which means the date/time that the move has been marked as in risk of excess weight) & if `excess_weight_acknowledged_at` is `nil` (which means the TOO has NOT marked that they are cool with it)
- If both of those are true, we are going to change the status of the move to `APPROVALS_REQUESTED` which should allow the TOO to see the `Review billable weight` modal which should populate the `excess_weight_acknowledged_at` value in the db

In addition to this, it was found that when the TOO requests to cancel a shipment that the move status was not being changed from `Approvals Requested` which would be an issue because they would see the move needs action, but no action is needed.
- updated the service object used when the TOO requests cancellation to consider if the move status should change back to `Approved`

### How to test

1. Access MM as Prime
2. Create a shipment for any move that is in `Move approved` status but don't include the estimated weight
3. Approve the shipment as TOO
4. Go back in as the prime and update the estimated weight to be either under/over the weight allowance
5. If over, the status of the move should change to `Approvals Requested` and the TOO should see it in their queue and need to review billable weight
6. If under, no action should be needed from the TOO and the status should remain as it was

1. Access MM as TOO
2. Manually change the status of a move that contains a shipment to `APPROVALS_REQUESTED`
3. This should now show up in the queue as `Approvals Requested`
4. Go in and request cancellation of a shipment via the MTO page
5. Status should now change back to `Approved` for the TOO when viewing the queue
